pkg/handlers/generic/mutation/controlplanevirtualip/providers/kubevip.go

```diff
@@ -39,39 +38,32 @@ var (
 //go:embed templates/configure-for-kube-vip.sh
 var configureForKubeVIPScript []byte
 
-type kubeVIPFromConfigMapProvider struct {
-	client client.Reader
-
-	configMapKey client.ObjectKey
+type kubeVIPFromKCPTemplateProvider struct {
+	template *controlplanev1.KubeadmControlPlaneTemplate
 }
 
-func NewKubeVIPFromConfigMapProvider(
-	cl client.Reader,
-	name, namespace string,
-) *kubeVIPFromConfigMapProvider {
-	return &kubeVIPFromConfigMapProvider{
-		client: cl,
-		configMapKey: client.ObjectKey{
-			Name:      name,
-			Namespace: namespace,
-		},
+func NewKubeVIPFromKCPTemplateProvider(
+	template *controlplanev1.KubeadmControlPlaneTemplate,
+) *kubeVIPFromKCPTemplateProvider {
+	return &kubeVIPFromKCPTemplateProvider{
+		template: template,
 	}
 }
 
-func (p *kubeVIPFromConfigMapProvider) Name() string {
+func (p *kubeVIPFromKCPTemplateProvider) Name() string {
 	return "kube-vip"
 }
 
 // GenerateFilesAndCommands returns files and pre/post kubeadm commands for kube-vip.
 // It reads kube-vip template from a ConfigMap and returns the content a File, templating the required variables.
```

Comment:
update the comment too.

Reply:
Thanks, done in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1062/commits/1aae2c6919ae4dce46d266cb2576992fd541d553